File: Programación Orientada a Objetos (PDOO)/Napakalaki(Ruby)/pkg/pruebaNapakalaki-0.0.1/lib/death_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "bad_consequence.rb"
require_relative "numeric_bad_consequence.rb"
module NapakalakiGame
class DeathBadConsequence < NumericBadConsequence
  attr_accessor :death
  def initialize(txt)
    super(txt,0,0,0)
    @death=true
  end
  
  def to_s
    "#{@text}\n Niveles perdidos: #{@levels}\n Muerto: #{@death}" 
  end
  
end

end

File: Programación Orientada a Objetos (PDOO)/Napakalaki(Java)/src/napakalaki/DeathBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author Josele
 */
public class DeathBadConsequence extends NumericBadConsequence{
    private boolean death;
    public DeathBadConsequence(String txt,boolean dead){
        super(txt,0,0,0);
        this.death=dead;
    }
    
    public boolean getDeath(){
        return death;
    }
    
    @Override
    public boolean isEmpty(){
        boolean isEmpty = false;   
        if(this.death==false){
                isEmpty = true;
        }
        return isEmpty;
    }
    
    public String toString(){
        return /*" \n\tTexto = " + text + */" \n\tLevels = " + Integer.toString(levels) +" \n\tDeath = " + Boolean.toString(death);
    }
}
